nautobot/extras/plugins/__init__.py

```diff
@@ -169,7 +169,7 @@ def ready(self):
         if secrets_providers is not None:
             for secrets_provider in secrets_providers:
                 register_secrets_provider(secrets_provider)
-            self.features["secrets_providers"] = sorted(secrets_providers)
+            self.features["secrets_providers"] = sorted(p() for p in secrets_providers)
```

Comment:
Note that this only instantiates the providers as stored in the `self.features` dict - the `register_secrets_provider` call above still just registers the class, not any specific instance thereof, and the SecretsProvider API (`get_value_for_secret`) is a classmethod, not an instance method - if we want SecretsProviders to be instantiated before using them, that needs to be part of a larger API change I think.

Reply:
This is the only way to get ordering to work since `__le__`, `__eq__`, et al. are instance methods. We might just decide that we don't need them to be sorted when being registered and then this problem goes away?

Reply:
Also class methods will still work on an instance. They just don't require it.